modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogSchemaTest.java

```diff
@@ -17,33 +17,160 @@
+
+        assertThat(latestCatalog().schemas(), hasSize(initialSchemasCount + 1));
+
+        CatalogCommand cmd = DropSchemaCommand.builder().name(TEST_SCHEMA).build();
+
+        assertThat(manager.execute(cmd), willCompleteSuccessfully());
+        assertThat(latestCatalog().schema(TEST_SCHEMA), nullValue());
+        assertThat(latestCatalog().schemas(), hasSize(initialSchemasCount));
+
+        assertThat(
+                manager.execute(DropSchemaCommand.builder().name(TEST_SCHEMA).build()),
+                willThrowFast(CatalogValidationException.class, "Schema with name 'S1' not found")
+        );
+    }
+
+    @Test
+    public void testDropDefaultSchemaIsAllowed() {
+        CatalogCommand cmd = DropSchemaCommand.builder().name(SqlCommon.DEFAULT_SCHEMA_NAME).build();
+
+        assertThat(manager.execute(cmd), willCompleteSuccessfully());
+        assertThat(latestCatalog().schema(SqlCommon.DEFAULT_SCHEMA_NAME), nullValue());
+
+        assertThat(
+                manager.execute(simpleTable("test")),
+                willThrowFast(CatalogValidationException.class, "Schema with name 'PUBLIC' not found")
+        );
+    }
+
+    @Test
+    public void testDropNonEmpty() {
+        CatalogCommand newSchemaCmd = CreateSchemaCommand.builder().name(TEST_SCHEMA).build();
+        CatalogCommand newTableCmd = newTableCommand("T1");
+        CatalogCommand idxCmd = newIndexCommand("T1", "I1");
+
+        assertThat(
+                manager.execute(List.of(newSchemaCmd, newTableCmd, idxCmd)),
+                willCompleteSuccessfully()
+        );
+
+        // RESTRICT
+        {
+            assertThat(
+                    manager.execute(DropSchemaCommand.builder().name(TEST_SCHEMA).build()),
+                    willThrowFast(CatalogValidationException.class, "Schema 'S1' is not empty. Use CASCADE to drop it anyway.")
+            );
+
+            CatalogSchemaDescriptor schemaDescriptor = latestCatalog().schema(TEST_SCHEMA);
+
+            assertThat(schemaDescriptor, is(notNullValue()));
+            assertThat(schemaDescriptor.tables().length, is(1));
+        }
+
+        // CASCADE
+        {
+            assertThat(latestCatalog().tables(), hasSize(1));
+            assertThat(latestCatalog().indexes(), hasSize(2));
+
+            CatalogCommand dropCmd = DropSchemaCommand.builder().name(TEST_SCHEMA).cascade(true).build();
+
+            assertThat(manager.execute(dropCmd), willCompleteSuccessfully());
+
+            assertThat(latestCatalog().schema(TEST_SCHEMA), nullValue());
+            assertThat(latestCatalog().tables(), hasSize(0));
+            assertThat(latestCatalog().indexes(), hasSize(0));
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            CatalogService.SYSTEM_SCHEMA_NAME,
+            CatalogService.INFORMATION_SCHEMA,
+            CatalogService.DEFINITION_SCHEMA
+    })
+    public void testDropSystemSchemaIsForbidden(String schemaName) {
```

Comment:
I would suggest to move all validation tests to `<CommanName>ValidationTest` class for consistency

Reply:
Done, added `CatalogSchemaValidationTest` (for create/drop commands)